src/compas/geometry/_intersections/intersections.py

```diff
@@ -383,6 +384,40 @@ def intersection_segment_plane(segment, plane, tol=1e-6):
     return None
 
 
+def intersection_polyline_plane(polyline, plane, expected_number_of_intersections=None, tol=1e-6):
+    """Calculate the intersection point of a plane with a polyline. Reduce expected_number_of_intersections to speed up.
+
+    Parameters
+    ----------
+    polyline : compas.geometry.Polyline
+        polyline to test intersection
+    plane : compas.Geometry.Plane
+        plane to compute intersection
+    tol : float, optional
+        A tolerance for membership verification.
+        Default is ``1e-6``.
+
+    Returns
+    -------
+    list of :class:`compas.geometry.Point`
+        if there are intersection points, return point(s) in a list
+    """
+    if not expected_number_of_intersections:
+        expected_number_of_intersections = len(polyline)
+    intersection_points = []
+    max_iter = 0
+    for segment in pairwise(polyline):
+        pt = intersection_segment_plane(segment, plane, tol)
+        if pt and max_iter < expected_number_of_intersections:
+            intersection_points.append(pt)
+            max_iter += 1
+        else:
+            break
```

Comment:
why is this not just
```python
for segment in pairwise(polyline):
    if len(intersections) == expected_number_of_intersections:
        break
    point = intersection_segment_plane(segment, plane, tol)
    if point:
        intersections.append(point)
```

Reply:
yes, you are completely wright. Way better as you indicated. Thank you!